### ISSUE TITLE
Ensure Math Is Parsed/Rendered Consistently

### ISSUE THREAD

**Issue by phaustin** [executablebooks/MyST-NB#126]
To anchor our choices in some concrete examples, I've pushed #125 
which creates https://phaustin.github.io/MyST-NB/equno.html from
https://github.com/phaustin/MyST-NB/blob/eqno/docs/examples/eqno_test/equno.md

Questions:  

1) do we want or need similar behviour between latex-rendered and html-rendered versions of the same math?
2) What's the best way to implement the behavior we expect from 1)

(prelimnary aside -- @choldgraf -- as the html shows, I'm not getting mystnb.js to actually run on the page and toggle processEnvironments=true -- can you confirm that using the myst environment from eqno_test/environment.yml does properly fomat the no-$$ equation?  I see msytnb.js in _build/_static but it doesn't appear to do anything.)

**Comment by phaustin** [comment]
@chrisjsewell to check my understanding -- when you say "write a plugin" would it be a python port of  this markdown-it mathjax extension?  https://www.npmjs.com/package/markdown-it-mathjax

**Comment by chrisjsewell** [comment]
Yep something like that 👍 , we only need a new plugin that handles the latex environments component though, not the dollars/brackets are already handled by the https://github.com/goessner/markdown-it-texmath port, see https://github.com/ExecutableBookProject/markdown-it-py/tree/master/markdown_it/extensions/texmath

**Comment by phaustin** [comment]
that would be great, have no strong opinion on katex vs. mathjax 3 -- will note that some of our math-heavy mathjax 2.7 jupyer notebooks take a full 30 seconds to render with a 2019 macbook pro, so moving on from 2.7 was something we were looking at.

**Comment by choldgraf** [comment]
Just a note that I had a conversation about katex vs. mathjax with @rowanc1 and he mentioned that katex was way (way, way, way) faster for HTML rendering

**Comment by phaustin** [comment]
yes, although mathjax 3 claims it's doubled it's rendering speed, I gues that's not way**4 though

**Comment by chrisjsewell** [comment]
Yeh I also have got the impression that katex is the way forward these days

**Comment by rowanc1** [comment]
Interesting, that advice was pre-mathjax 3, which looks a lot better!

There are some detailed comparisons here:
https://www.intmath.com/cg5/katex-mathjax-comparison.php

Timings for me were: 314 vs 1341 vs 460 (katex, mathjax 2, mathjax 3)

There is a flicker on the mathjax loading which isn't there on katex, I think it is the font loading.

**Comment by choldgraf** [comment]
Interesting - sounds like speed isn't very different between katex and mathjax 3...

**Comment by phaustin** [comment]
One  plus for mathjax is more complete coverage of AMSmath environments -- not suprising since AMS sponsors it:  https://www.mathjax.org/#sponsors  -- but not a good look to brag about your "rich API" and send people to https://docs.mathjax.org/en/latest/api/index.html

**Comment by phaustin** [comment]
Not sure whether this should be a separate issue or not:  it would  be healthy for the notebook ecosystem if the myst equation/latex-environment syntax was round-trippable with bookdown.  Our 1st year datascience textook is currently in R/Rmd and formatted with bookdown, we'll be creating a parallel version in myst/python this year.  It's likely this will expand to upper year course modules shared by biology/stats (Rmd) and earth sciences/physics/math/cpsc (myst).  pagedown is also a good marker for  a feature -set, because it's been in use for about 4 years now and presumably some rough edges have been smoothed out:  https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html

**Comment by choldgraf** [comment]
I think that is probably a separate issue since it touches a lot more than just math. We've definitely chosen the MyST-markdown syntax to take strong inspiration from bookdown, so visually they are quite similar. I think we could create an issue that just had a list of bookdown extended markdown features and their respective MyST counterparts

**Comment by phaustin** [comment]
will do -- for the record for this issue, the pertinent features of bookdown equation numbering for html rendering appears to be:

- only labeled equations are numbered
- the label appears inside the latex environment
- environments without a label should use `\begin{xxx*}`
   to insure the latex-rendered version is unnumbered in the pdf
- latex enivironments can appear on their own or wrapped in $$ delimiters

**Comment by choldgraf** [comment]
Hey all - I am not sure what's the actionable next step on this issue. Can somebody clarify (and ideally post something in the top comment?)

**Comment by chrisjsewell** [comment]
> Hey all - I am not sure what's the actionable next step on this issue. Can somebody clarify (and ideally post something in the top comment?)

- [ ] Add extension to markdown-it-py: https://github.com/executablebooks/markdown-it-py/pull/12
- [ ] Add render method to myst-parser, similar to https://github.com/executablebooks/MyST-Parser/blob/a084975f02c0b4a9141f75878f78b48afa9f9b5a/myst_parser/docutils_renderer.py#L551
  - an open question is how to handle internal `\label` and `\tag`
- [ ] Add a sphinx option to load the extension (by default off?) within https://github.com/executablebooks/MyST-Parser/blob/a084975f02c0b4a9141f75878f78b48afa9f9b5a/myst_parser/sphinx_parser.py#L196 and https://github.com/executablebooks/MyST-NB/blob/5e6e8f01cdd612bb766302dc40327ca2aa680b1f/myst_nb/parser.py#L89

**Comment by AakashGfude** [comment]
@chrisjsewell @choldgraf @mmcky  as this is something which needs to be implemented for latex style math to be rendered in pdf I believe. As mentioned in this comment:-  https://github.com/executablebooks/jupyter-book/pull/586#issuecomment-664072602 . If no one else is working on it, should I pursue this feature?  What should be the actionable next steps?

**Comment by chrisjsewell** [comment]
Thats ok @AakashGfude I was going to pick this up soon (I've just forwarded you an email of some discussions about my plans, which I'll turn in to a GitHub project at some point)
One thing you might want to look at is https://github.com/executablebooks/MyST-Parser/issues/183. This would allow you to get more acquainted with the myst-parser code, and once this issue is closed it would also open up the possibility to consider https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-latex_theme
